### PR TITLE
Combing MainCLI and Main into single file. Update Args with headless …

### PR DESCRIPTION
**Pull request by iwodder**
@lupino3 ready for a first check on this PR. The stock JDK SplashScreen class didn't work for this, so I created a custom one based on some searching. This eliminates the splash popping up briefly when running in headless mode but with a GUI enabled. Tested this using a SSH session for headless. I see one clean up item where a class which is imported is explicitly qualified. Will fix after feedback! Thanks!

### CODE REVIEW

**Comment by lupino3** [comment]
Thanks a lot, I'll start reviewing now! In the meantime could you please fix the CI by removing the step to build the CLI JAR from here: https://github.com/EduMIPS64/edumips64/blob/master/.github/workflows/gradle.yml#L50

File: docs/user/en/src/user-interface.rst [hunk 1]
```diff
@@ -268,6 +268,8 @@ be used in the same way.
 
 * `-d (--debug)` enters Debug mode
 
+* `-hl (--headless)` Runs Edu MIPS 64 in headless mode (no gui)
```
Comment (lupino3) [inline docs/user/en/src/user-interface.rst:271]: Minor suggestion: EduMIPS64 is spelled as a single word.
```suggestion
* `-hl (--headless)` Runs EduMIPS64 in headless mode (no gui)
```

File: docs/user/it/src/user-interface.rst [hunk 2]
```diff
@@ -266,7 +266,7 @@ del presente documento.
 
 Opzioni da riga di comando
 --------------------------
-Sono disponibili tre opzioni da linea di comando. Esse sono descritte di
+Quattro disponibili tre opzioni da linea di comando. Esse sono descritte di
```
Comment (lupino3) [inline docs/user/it/src/user-interface.rst:269]:
```suggestion
Sono disponibili quattro opzioni da linea di comando. Esse sono descritte di
```

File: docs/user/it/src/user-interface.rst [hunk 3]
```diff
@@ -282,6 +282,8 @@ esteso possono essere utilizzati indifferentemente.
 
 * `-d (--debug)` attiva la modalità di debugging.
 
+* `-hl (--headless)` Esegue Edu MIPS 64 in modalità headless (senza interfaccia grafica)
```
Comment (lupino3) [inline docs/user/it/src/user-interface.rst:285]:
```suggestion
* `-hl (--headless)` Esegue EduMIPS64 in modalità headless (senza interfaccia grafica)
```

File: src/main/java/org/edumips64/Main.java [hunk 4]
```diff
@@ -145,9 +146,23 @@ public static void main(String args[]) {
...
+    if (cliArgs.isHeadless()) {
+      startCli(mm.configStore, cliArgs);
+    } else {
+      checkAndStartGui(mm, cliArgs);
+    }
+  }
+
+  private static void checkAndStartGui(Main mm, Args cliArgs) {
+    if (GraphicsEnvironment.isHeadless()) {
+      log.log(Level.SEVERE, "Cannot start GUI in a headless environment.");
```
Comment (lupino3) [inline src/main/java/org/edumips64/Main.java:160]: This error message should probably be localized.

File: src/main/java/org/edumips64/ui/swing/SplashScreen.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
+package org.edumips64.ui.swing;
```
Comment (lupino3) [inline src/main/java/org/edumips64/ui/swing/SplashScreen.java:1]: Can you please add the standard header (copyright, license, author) to this file?

File: src/main/resources/Messages.properties [hunk 6]
```diff
@@ -2,4 +2,5 @@ file=filename - opens the specified file
 debug=activates debug mode
 help=prints this help message
 reset=resets the stored preferences
-version=prints the version
\ No newline at end of file
+version=prints the version
+headless=Runs Edu MIPS 64 in headless mode (no gui)
```
Comment (lupino3) [inline src/main/resources/Messages.properties:6]:
```suggestion
headless=Runs EduMIPS64 in headless mode (no gui)
```

File: src/main/resources/Messages_it.properties [hunk 7]
```diff
@@ -2,4 +2,5 @@ file=apre il file specificato
 debug=attiva la modalità di debug
 help=stampa questo messaggio
 reset=elimina le preferenze memorizzate
-version=stampa la versione
\ No newline at end of file
+version=stampa la versione
+headless=Esegue Edu MIPS 64 in modalità headless (senza interfaccia grafica)
```
Comment (lupino3) [inline src/main/resources/Messages_it.properties:6]:
```suggestion
headless=Esegue EduMIPS64 in modalità headless (senza interfaccia grafica)
```

**Review by lupino3 — CHANGES_REQUESTED** [review]
Overall looks good, thanks! Once you fix the CI I'll be able to test the JAR locally (I don't have a Java dev environment available right now).

Thanks a lot for this new feature!

File: src/main/java/org/edumips64/Main.java (hunk 4 above)
Comment (iwodder) [reply src/main/java/org/edumips64/Main.java]: On it!

File: src/main/java/org/edumips64/ui/swing/SplashScreen.java (hunk 5 above)
Comment (iwodder) [reply src/main/java/org/edumips64/ui/swing/SplashScreen.java:23]: Done!

**Comment by lupino3** [comment]
Tested it locally and works great! Let me know when all comments are addressed and I'll merge the PR!

I'll try to do a release in the upcoming days/weeks (on leave now and have limited available time to do it). Thanks a lot!

File: src/main/java/org/edumips64/ui/swing/SplashScreen.java [hunk 8]
```diff
@@ -0,0 +1,71 @@
+/* SplashScreen.java
+ *
+ * SplashScreen for EduMIPS64
+ * (c) 2006 Andrea Spadaccini, Antonella Scandura, Vanni Rizzo
```
Comment (lupino3) [inline src/main/java/org/edumips64/ui/swing/SplashScreen.java:4]: You should put your name (or username) here and the year 2020.

Comment (iwodder) [reply src/main/java/org/edumips64/ui/swing/SplashScreen.java]: Done, this one and the other two pending have been corrected.

**Review by lupino3 — APPROVED** [review]
LGTM, thank you!

**Pull request merged**